src/node_lib.h

```diff
@@ -45,25 +57,30 @@ namespace node::lib {
     /*
     Processes the pending event queue of a *running* Node.js engine once.
     */
-    bool ProcessEvents();
+    NODE_EXTERN bool ProcessEvents();
 
     /*
-    Starts the execution of the Node.js main loop, which processes any events in JavaScript.
+    Starts the execution of the Node.js event loop, which processes any events in JavaScript.
     Additionally, the given callback will be executed once per main loop run.
     *Important*: Call `initialize()` before using this method.
     */
-    bool Run(const RunUserLoop & callback);
+    NODE_EXTERN void RunEventLoop(const RunUserLoop & callback);
 
 
     /*********************************************************
      * Stop Node.js engine
      *********************************************************/
 
     /*
-    Stops the *running* Node.js engine.
+    Sends termination event into event queue and runs event queue, until all events have been handled.
     */
-    bool Terminate();
+    NODE_EXTERN void Terminate();
+
 
+    /*
+    Stops the *running* Node.js engine. Clears all events and puts Node.js into idle.

```

Comment:
Wrong documentation. This shuts the Node.js engine down; not put it into "idle".

Reply:
The idea here was to be able to restart the event loop without having to reinitialize the entire node engine. Additionally, we wanted to add a Deinitialize() function that completely destroys node's state, resulting in having to call Initialize() again to be able to use node again, with a new, clean context. Adding Deinitialize() in header file right now.